notebook/notebook/notebook.py

```diff
@@ -131,10 +171,153 @@ def external_url_for(path):
     return url + path
 
 
+def forbidden():
+    return 'Forbidden', 403
+
+
 @app.route('/healthcheck')
 def healthcheck():
     return '', 200
 
+##### New notebook methods meant for consumption from agnostic client#####
+###### Methods that do not call the auth-gateway /verify method must not be directly public #######
+@app.route('/verify/<svc_name>/', methods=['GET'])
+def validate_get(svc_name):
+    access_token = request.args.get('authorization')
+    token = request.args.get('token')
+
+    if not access_token:
+        return forbidden()
+
+    resp = requests.get(f'http://{AUTH_GATEWAY}/verify',
+                        headers={'Authorization': f'Bearer {access_token}'})
+
+    if resp.status_code != 200:
+        return forbidden()
+
+    user_id = resp.headers.get('User')
+
+    if not user_id:
+        return forbidden()
+
+    try:
+        mysql_conn = pymysql.connect(
+        host=IP, port=PORT, user=USER, passwd=PASS, db=DB)
+
+        with mysql_conn.cursor() as cur:
+            cur.execute(f"SELECT svc_name FROM sessions WHERE user_id='{user_id}' AND token='{token}'")
+
+            rows = cur.fetchall()
+
+            if len(rows) != 1 or rows[0][0] != svc_name:
+                return forbidden()
+
+            return '', 200
+    except:
```

Comment:
these blocks are not necessary, flask will issue a 500 if any exception occurs and print the stack trace to the log.

Reply:
A 500 message is reserved for programmatic errors. I want user-space/operational errors, so that the client can distinguish them

Reply:
Got it; misunderstood your initial comment

Reply:
I do prefer explicit error handling. When I look at a function I don't expect it to catch errors for me, unless that's the language specification. It also gives us a chance to send more useful error messages, and by not relying on Flask's implementation will have code that is more portable to other server environments that may not wrap everything in a try/catch.

Reply:
I hear your perspective on this. I don't like higher-order functions that have many responsibilities. They are hard to follow. app.route should do one thing, match a handler to an endpoint.